src/lib/seam/connect/routes/client-sessions.ts

```diff
@@ -0,0 +1,159 @@
+  SeamHttpInvalidOptionsError,
+  type SeamHttpOptions,
+  type SeamHttpOptionsWithApiKey,
+  type SeamHttpOptionsWithClient,
+  type SeamHttpOptionsWithClientSessionToken,
+} from 'lib/seam/connect/client-options.js'
+import { parseOptions } from 'lib/seam/connect/parse-options.js'
+
+export class SeamHttpClientSessions {
+  client: Axios
+
+  constructor(apiKeyOrOptions: string | SeamHttpOptions) {
+    const options = parseOptions(apiKeyOrOptions)
+    this.client = createAxiosClient(options)
+  }
+
+  static fromClient(
+    client: SeamHttpOptionsWithClient['client'],
+    options: Omit<SeamHttpOptionsWithClient, 'client'> = {},
+  ): SeamHttpClientSessions {
+    const opts = { ...options, client }
+    if (!isSeamHttpOptionsWithClient(opts)) {
+      throw new SeamHttpInvalidOptionsError('Missing client')
+    }
+    return new SeamHttpClientSessions(opts)
+  }
+
+  static fromApiKey(
+    apiKey: SeamHttpOptionsWithApiKey['apiKey'],
+    options: Omit<SeamHttpOptionsWithApiKey, 'apiKey'> = {},
+  ): SeamHttpClientSessions {
+    const opts = { ...options, apiKey }
+    if (!isSeamHttpOptionsWithApiKey(opts)) {
+      throw new SeamHttpInvalidOptionsError('Missing apiKey')
+    }
+    return new SeamHttpClientSessions(opts)
+  }
+
+  static fromClientSessionToken(
+    clientSessionToken: SeamHttpOptionsWithClientSessionToken['clientSessionToken'],
+    options: Omit<
+      SeamHttpOptionsWithClientSessionToken,
+      'clientSessionToken'
+    > = {},
+  ): SeamHttpClientSessions {
+    const opts = { ...options, clientSessionToken }
+    if (!isSeamHttpOptionsWithClientSessionToken(opts)) {
+      throw new SeamHttpInvalidOptionsError('Missing clientSessionToken')
+    }
+    return new SeamHttpClientSessions(opts)
+  }
+
+  async create(
+    body: ClientSessionsCreateBody,
+  ): Promise<ClientSessionsCreateResponse['client_session']> {
+    const { data } = await this.client.request<ClientSessionsCreateResponse>({
+      url: '/client_sessions/create',
+      method: 'post',
+      data: body,
+    })
+    return data.client_session
+  }
+
+  async delete(body: ClientSessionsDeleteBody): Promise<void> {
+    await this.client.request<ClientSessionsDeleteResponse>({
+      url: '/client_sessions/delete',
+      method: 'post',
+      data: body,
+    })
+  }
+
+  async get(
+    body: ClientSessionsGetBody,
+  ): Promise<ClientSessionsGetResponse['client_session']> {
+    const { data } = await this.client.request<ClientSessionsGetResponse>({
+      url: '/client_sessions/get',
+      method: 'post',
+      data: body,
+    })
+    return data.client_session
+  }
+
+  async grantAccess(
+    body: ClientSessionsGrantAccessBody,
+  ): Promise<ClientSessionsGrantAccessResponse['client_session']> {
+    const { data } =
+      await this.client.request<ClientSessionsGrantAccessResponse>({
+        url: '/client_sessions/grant_access',
+        method: 'post',
+        data: body,
+      })
+    return data.client_session
```

Comment:
Should this return the client session?

Reply:
I think this is correct since if the user identifier key is used to update the client session, you will want to get the client session that matched the key back.